lib/hammer_cli_foreman/compute_resource.rb

```diff
@@ -70,24 +72,15 @@ def print_data(data)
 
     class CreateCommand < HammerCLIForeman::CreateCommand
 
-      VALIDATION_PER_PROVIDER = {
-        'libvirt'   => [:option_url],
-        'ovirt'     => [:option_url, :option_user, :option_password, :option_datacenter],
-        'openstack' => [:option_url, :option_user, :option_password],
-        'racksapce' => [:option_url],
-        'ec2'       => [:option_region, :option_user, :option_password],
-        'vmware'    => [:option_user, :option_password, :option_datacenter, :option_server]
-      }
-
       success_message _("Compute resource created.")
       failure_message _("Could not create the compute resource")
 
       build_options
 
       validate_options do
-        required_options = (VALIDATION_PER_PROVIDER[
+        required_options = (::HammerCLIForeman.compute_resources[
           option(:option_provider).value.to_s.downcase
-        ] || []).unshift(:option_name, :option_provider).uniq
+        ] || []).compute_resource_creation_validation
```

Comment:
can we make `compute_resource_creation_validation` more generic? E.g. 
```ruby 
def mandatory_attributes; [:name, :provider]; end
```
and use it like 
```ruby
... .mandatory_attributes.map { |attr| "option_#{attr}".to_sym }
```

Reply:
I see now the name I suggested may lead to confusion with host attributes and understand why you chose the name. Maybe `mandatory_resource_options` could help to avoid the `validation` in the name which is suggesting it is validating something?

Reply:
@mbacovsky
 can you explain how the mandatory_resource_options could help avoid the validation in the name?
thanks.

Reply:
'mandatory_resource_options' would be new name for `compute_resource_creation_validation`. 
In the base class you'd have
```ruby
def mandatory_resource_options; [:name, :provider]; end
```
in the libvirt it would be
```ruby
def mandatory_resource_options; super + [:url]; end
```
etc. And in the command it would be
```ruby
      validate_options do
         required_options = (::HammerCLIForeman.compute_resources[
￼             option(:option_provider).value.to_s.downcase		￼           
￼           ] || []).mandatory_resource_options.map { |attr| "option_#{attr}".to_sym }
         all(*required_options).required
￼       end
```
Makes sense now?